.ci/Jenkinsfile

```
@@ -502,3 +577,9 @@ def generateStepForWindows(Map params = [:]){
     sh label: 'Stop services', script: ".ci/scripts/windows/stop-test.sh ${version}"
   }
 }
+
+def isValidReleaseVer(String ver){
+    if (ver == "0.0.0") {
+        error("Must supply version number to release")
+    }
```

The version could be a typo, if semversion in the format, what about using a few more validations:
- no empty values
- no default value
- no sem version format (major.minor.patch)

NOTE: I have not tested the below snippet


```suggestion
    if (!ver?.trim() || ver == "0.0.0" || !(ver ==~ /(\d+)\.(\d+).(\d+)/) ) {
        error("Must supply version number to release")
    }
```

---

I'd limit the string start and end using `^` and `$` in regular expressions. For the official semver regex, see https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string - but this is if you also want to support the build metadata, which I'm not sure if the automated release needs

---

Groovy is a bit tricky, and actually. you got a good point, the `==~` operator is the exact match based, therefore `^` and `$` are not required. If it is not intuitive, then we could force the match with the find operator `=~` instead and the `^` and `$`.